src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -473,27 +470,33 @@ private long getOffset()
 
     private void flush(PartitionBuffer buff)
     {
-      if (buff == null || buff.isEmpty())
-      {
-        return;
-      }
+      flusherExecutor.submit(
```

Comment:
detach this thread. Note that this thread does not contain while loop so it won't leak. Async here because in the insert we check all pipes whether they need to flush or not. If it happens that all of them need to be flushed, synced flush will block insert for a long time. Therefore do an async flush.